### PR TITLE
[MINOR] SDK-1789 : Update unit test tools

### PR DESCRIPTION
**Pull request by JagadeeshKaricherla-branch**
# Pull Request Template

## Description

- Update testing library, scripts - mocha and add nyc code coverage
- Update test.html and remove all sauce labs instances
- update test scripts - nycrc, mocharc
- update make file

Fixes # (issue) : SDK-1789

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit test
- [x] Integration test

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

## Mentions: 
List the person or team responsible for reviewing proposed changes.

cc @BranchMetrics/saas-sdk-devs for visibility.

### CODE REVIEW

File: test/1_utils.js [hunk 1]
```diff
@@ -420,7 +424,7 @@ describe('utils', function() {
 				'should return an object type'
 			);
 		});
-		it('should return OG tags', function() {
+		it.skip('should return OG tags', function() {
```
Comment (gdeluna-branch) [inline test/1_utils.js:427]: For tests that are skipped, what work is needed to reenable them?

File: package-lock.json [hunk 2]
```diff
@@ -4916,222 +4216,162 @@
...
-			"engines": {
-				"node": ">=0.10.0"
 			}
 		},
 		"node_modules/stable": {
 			"version": "0.1.8",
 			"resolved": "https://registry.npmjs.org/stable/-/stable-0.1.8.tgz",
 			"integrity": "sha512-ji9qxRnOVfcuLDySj9qzhGSEFVobyt1kIOSkj1qZzYLzq7Tos/oUUWvotUPQLlrsidqsK6tBH89Bc9kL5zHA6w==",
 			"deprecated": "Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility",
+<<<<<<< HEAD
```
Comment (gdeluna-branch) [inline package-lock.json:4226]: Looks like there are merge conflicts in the `package-lock.json`. should be able to just regenerate

File: test/1_utils.js [hunk 3]
```diff
@@ -1,15 +1,19 @@
 'use strict';
-
+/*jshint -W079 */
+var sinon = require('sinon');
```
Comment (gdeluna-branch) [inline test/1_utils.js:3]: I'm curious what else is defining `sinon`

File: test/1_utils.js (hunk 1 above)
Comment (JagadeeshKaricherla-branch) [reply test/1_utils.js:427]: fix and re-enable.
And some of these tests are old and need to be reviewed and removed.

File: test/1_utils.js (hunk 3 above)
Comment (JagadeeshKaricherla-branch) [reply test/1_utils.js:3]: this is a rule conflict between google closure compiler and jshint rules.
In the near future, we are going to replace jshint with eslint and this re-definition issue won’t  be present.

**Review by gdeluna-branch — APPROVED** [review]

**Pull request merged**